main/src/ca/uwaterloo/flix/api/lsp/Indexer.scala

```diff
@@ -329,6 +329,9 @@ object Indexer {
     case Expression.Upcast(exp, tpe, _) =>
       visitExp(exp) ++ visitType(tpe) ++ Index.occurrenceOf(exp0)
 
+    case Expression.Supercast(exp, tpe, _) =>
+      visitExp(exp) ++ visitType(tpe) ++ Index.occurrenceOf(exp0)
```

Comment:
The indexer should only visit types that are _visible_. Since the type is not written in the source code here, it shouldn't be visited. (Same for upcast above)